apps/modernization-ui/src/apps/page-builder/page/management/edit/subsection/SubsectionHeader.tsx

```diff
@@ -18,19 +18,13 @@ type Props = {
 
 export const SubsectionHeader = ({ name, id, questionCount, isExpanded, onExpandedChange, onAddQuestion }: Props) => {
     const addStaticElementModalRef = useRef<ModalRef>(null);
-    const queListModalRef = useRef<ModalRef>(null);
+    const addQuestionModalRef = useRef<ModalRef>(null);
     const renderQuestionListModal = () => (
         <>
-            <ModalToggleButton className="add-btn" outline modalRef={queListModalRef}>
+            <ModalToggleButton className="add-btn" outline modalRef={addQuestionModalRef}>
                 Add Question
             </ModalToggleButton>
-            <ModalComponent
-                isLarge
-                forceAction={false}
-                modalRef={queListModalRef}
-                modalHeading={'Add question'}
-                modalBody={<QuestionLibrary hideTabs onAddQuestion={onAddQuestion} modalRef={queListModalRef} />}
-            />
+            <AddQuestionModal onAddQuestion={onAddQuestion} modalRef={addQuestionModalRef} />
```

Comment:
Issue: Adding the `modalRef` to the `SubsectionHeader` component will create 1 `AddQuestionModal` per `Subsection`. It would be better to have the `AddQuestionModal` in the `PageContent` component and "pass up" the request to create a new question.

This is especially true since the `AddQuestionModal` makes API calls to fetch data to display in drop downs. With this implementation each API call would be sent once per `Subsection`.